library/cartridge_set_package_info.py

```diff
@@ -94,29 +94,61 @@ def get_deb_info(package_path):
 
     return {
         'name': package_name,
-        'tnt_version': tnt_version
+        'tnt_version': tnt_version,
+    }
+
+
+def get_tgz_info(package_path, app_name):
+    with tarfile.open(package_path) as tar:
+        try:
+            tar.getmember(app_name)
+        except KeyError:
+            raise Exception("Package should contain %s directory" % app_name)
+
+        version_file_path = os.path.join(app_name, 'VERSION')
+        try:
+            member = tar.getmember(version_file_path)
+        except KeyError:
+            raise Exception("Package should contain %s file" % version_file_path)
+
+        version_file = tar.extractfile(member)
+        version_file_lines = version_file.readlines()
```

Comment:
I'm not sure, but maybe it's necessary to close stream:
```suggestion
        with tar.extractfile(member) as version_file:
            version_file_lines = version_file.readlines()
```

Reply:
It doesn't work since `version_file` doesn't have `close` method.